### PR TITLE
feat(file): Use new writer

### PR DESCRIPTION
**Pull request by disq**
Example on how to utilize https://github.com/cloudquery/plugin-sdk/pull/870
Blocked by https://github.com/cloudquery/filetypes/pull/154

Part of https://github.com/cloudquery/cloudquery/issues/10591

TODO: After this is merged we can add max file size or compressed streaming settings.

### CODE REVIEW

File: plugins/destination/file/go.mod [hunk 1]
```diff
@@ -15,6 +15,12 @@ require (
 // TODO: remove once all updates are merged
 replace github.com/apache/arrow/go/v13 => github.com/cloudquery/arrow/go/v13 v13.0.0-20230521112802-adef07d4bbaa
 
+replace github.com/cloudquery/plugin-sdk/v3 => github.com/disq/plugin-sdk/v3 v3.0.0-20230525203908-a2e923d19b8e
+
+//replace github.com/cloudquery/plugin-sdk/v3 => ../../../../plugin-sdk
+
+replace github.com/cloudquery/filetypes/v3 => github.com/cloudquery/filetypes/v3 v3.1.1-0.20230525203919-7a1d60739423
```
Comment (candiduslynx) [inline plugins/destination/file/go.mod:22]: IMO this can already be removed

File: plugins/destination/file/go.mod [hunk 2]
```diff
@@ -15,6 +15,12 @@ require (
 // TODO: remove once all updates are merged
 replace github.com/apache/arrow/go/v13 => github.com/cloudquery/arrow/go/v13 v13.0.0-20230521112802-adef07d4bbaa
 
+replace github.com/cloudquery/plugin-sdk/v3 => github.com/disq/plugin-sdk/v3 v3.0.0-20230525203908-a2e923d19b8e
```
Comment (candiduslynx) [inline plugins/destination/file/go.mod:18]: is the required PR merged?

File: plugins/destination/file/client/write.go [hunk 3]
```diff
@@ -26,9 +34,45 @@ func (c *Client) WriteTableBatch(ctx context.Context, table *schema.Table, data
...
-	defer f.Close()
 
-	return c.Client.WriteTableBatchFile(f, table, data)
+	h, err := c.Client.WriteHeader(f, table)
+	if err != nil {
+		return err
+	}
+
+	key := mapKey(sourceSpec, table)
+	c.tableFiles[key] = writer{
```
Comment (candiduslynx) [inline plugins/destination/file/client/write.go:44]:
```suggestion
	c.tableFiles[mapKey(sourceSpec, table)] = writer{
```

File: plugins/destination/file/client/write.go [hunk 4]
```diff
@@ -26,9 +34,45 @@ func (c *Client) WriteTableBatch(ctx context.Context, table *schema.Table, data
...
+
+	key := mapKey(sourceSpec, table)
+	err := c.tableFiles[key].h.WriteFooter()
+	err2 := c.tableFiles[key].f.Close()
+	delete(c.tableFiles, key)
+	if err != nil {
+		return err
+	}
+	return err2
+}
```
Comment (candiduslynx) [inline plugins/destination/file/client/write.go:67]:
```suggestion
	key := mapKey(sourceSpec, table)
	tFile := c.tableFiles[key]
	delete(c.tableFiles, key)

	err := tFile.h.WriteFooter()
	err2 := tFile.f.Close()
	if err != nil {
		return err
	}
	return err2
}
```

File: plugins/destination/file/client/write.go [hunk 5]
```diff
@@ -26,9 +34,45 @@ func (c *Client) WriteTableBatch(ctx context.Context, table *schema.Table, data
...
+	}
+	c.lastSpec = sourceSpec
+
+	return nil
+}
+
+func (c *Client) CloseTable(_ context.Context, sourceSpec specs.Source, table *schema.Table, _ time.Time) error {
+	c.logger.Debug().Str("source", sourceSpec.Name).Str("table", table.Name).Msg("CloseTable")
+
+	c.tableFilesMu.Lock()
```
Comment (candiduslynx) [inline plugins/destination/file/client/write.go:56]: can sync.Map be used instead (we can use https://pkg.go.dev/sync#Map.LoadAndDelete for this one)?

**Review by candiduslynx — CHANGES_REQUESTED** [review]
I suggest moving to https://pkg.go.dev/sync#Map

**Comment by disq** [comment]
Closing for sdk v4

**Pull request closed without merging**